lib/public_key/include/public_key.hrl

```diff
@@ -41,47 +41,59 @@
 		 {valid, UserState}
 	 end, []}).
 
--record(path_validation_state, {
-	  valid_policy_tree,
-	  explicit_policy,
-	  inhibit_any_policy,
-	  policy_mapping,
-	  cert_num,
-	  last_cert = false,
-	  permitted_subtrees = no_constraints, %% Name constraints
-	  excluded_subtrees = [],      %% Name constraints   
-	  working_public_key_algorithm,
-	  working_public_key,
-	  working_public_key_parameters,
-	  working_issuer_name,
-	  max_path_length,
-	  verify_fun,
-	  user_state
-	 }).
+-record(path_validation_state,
+        {
+         valid_policy_tree,
+         user_inital_policy_set,
+         explicit_policy,
+         inhibit_any_policy,
+         inhibit_policy_mapping,
+         policy_mapping_ext,
+         policy_constraint_ext,
+         policy_inhibit_ext,
+         present_policy_ext,
```

Comment:
- thanks for explaining
- i think it would be easier to read if it is `policy_ext_present` - having `_ext` at the end is used for fields storing extension in other places
- i don't understand `current_any_policy_qualifiers` - maybe I will understand later. if `current` is meaning same thing as `working` then it should be the latter ... but I don't know it yet
- i don't understand `...**any_policy_**qualifiers` ... at some processing point value is stored in variable AnyQualifiers and later in qualifier_set inside a map. what is meant by `any_policy` qualifiers ...

Reply:
The certificate policies extension:

```
PolicyInformation ::= SEQUENCE {
     policyIdentifier   CertPolicyId,
     policyQualifiers   SEQUENCE SIZE (1..MAX) OF
             PolicyQualifierInfo OPTIONAL }
```

can contain ?anyPolicy and quilfiers for them, that will be handled in Section 6.1.3 d
step 2!